include/kodlab_mjbots_sdk/behavior.h

```diff
@@ -137,6 +140,27 @@ class Behavior
     return true;
   }
 
+  /**
+   * @brief Processes input data, if available
+   * @details This method should be implemented by child classes wishing to
+   * handle input processing internally. For supported LCM behavior inputs,
+   * please inherit `IOBehavior`.
+   * @note If using a behavior manager, this method is called every update
+   * cycle, and should implement checking to determine if a new input is
```

Comment:
I was mostly just trying to keep these additional methods available to the user if desired.  There is some world where additional input is desired (event-based or otherwise), and these methods can be overridden to handle those.  

Since we are working exclusively with LCM, all I/O should be done with a `BehaviorIO` child, and these functions likely will not be used otherwise.  That said, I don't want to sacrifice generality.  If we go the direction of making these private and making the classes friends to `BehaviorManager`, then we should go all the way and do that for all of the `Behavior` methods that are called by the `behaviorManager`.

Reply:
In that case I would just clarify  the should. If they are implementing something event based it is on them to do that if they are just using it as another update function that is there prerogative too.  So I might change it to 
`@note If using behavior manager, this method is called every update cycle after the Update function.`

Reply:
I dont know if it actually after or before, but we should clarify that

Reply:
Clarified in 57cf31c6c2d3f617d25dde54873399c802c2f84c